src/Symfony/Bundle/WebProfilerBundle/Tests/EventListener/MockedStreamedResponse.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\WebProfilerBundle\Tests\EventListener;
+
+use Symfony\Component\HttpFoundation\StreamedResponse;
+
+class MockedStreamedResponse extends StreamedResponse
+{
+    public function getContent(): string|false
```

Comment:
I would not override the `getContent` method of StreamedResponse for that, as it means this MockedStreamedResponse does not actually behave like a StreamedResponse (in which you _cannot_ call `getContent` to get the actual content), which would potentially hide bugs in tests.

Instead, I would suggest writing a separate test covering the injection of the toolbar in a streamed response, which would perform this output buffering in the test (using `$response->sendContent()`).

Reply:
You're right and indeed I missed a bug on StreamedResponse [here](https://github.com/symfony/symfony/pull/58789/files#diff-af937a8407173a2c929cd87fd9394095d21199f8c2cc1627eaeaa7935db0532cR109).

Let me know if it's ok for you.